Added performance impact statement for enableAnalysis flag (#11121)
istio/istio.io: content/en/docs/ops/diagnostic-tools/istioctl-analyze/index.md

$ kubectl annotate deployment my-deployment galley.istio.io/analyze-suppress=IST0107,IST0002
{{< /text >}}

> ### Performance impact when turning on validation

Reviewer: ```suggestion
### Performance impact when enabling validation
```

> Set the flag `enableAnalysis` to true when install Istio or upgrade your Istio control plane will allow istiod to analyze Istio resources and update their status fields. Istiod runs analyzers in the background every 10 seconds by default to inspect Istio resources for possible issues, this work will take CPU, memory resources to complete, thus the CPU and memory consumption and performance impact shall be evaluated, the [work](https://github.com/litong01/multicluster-istio/blob/main/perf/charts/README.md) was an attempt to do that evaluation. Based on that work, when `enableAnalysis` flag is turned on, comparing to when the flag is turned off, the memory consumptions can increase up to 34%, number of go routine calls can increase up to 12%, however the CPU usages stay relatively flat, in most of the tests performed, the CPU consumption actually drops some. Please follow the link above to see the details on how the tests were designed and performed. The performance impact in your production environment may produce different numbers in terms of the memory, CPU consumptions.

Reviewer: ```suggestion
Set the flag `enableAnalysis` to true when installing or upgrading your Istio control plane will allow istiod to analyze Istio resources and update their status fields. Istiod runs analyzers in the background every 10 seconds, by default, to inspect Istio resources for possible issues. This work will take CPU and memory resources to complete, thus the CPU and memory consumption and performance impact shall be evaluated. The [work](https://github.com/litong01/multicluster-istio/blob/main/perf/charts/README.md) was an attempt to do that evaluation. Based on that work, when `enableAnalysis` flag is enabled, compared to when the flag is disabled, the memory consumptions can increase up to 34%, the number of go routine calls can increase up to 12%, however the CPU usages stay relatively flat. In most of the tests performed, the CPU consumption actually drops slightly.. Please follow the link above to see the details on how the tests were designed and performed. The performance impact in your production environment may produce different numbers in terms of the memoryand CPU consumption.
```

> Due to the performance impact, when this flag is turned on, you probably can also utilize parameter[`PILOT_ANALYSIS_INTERVAL`](/docs/reference/commands/istioctl/) (available since 1.13.2) to balance how much resources you are willing to spend for validation vs how soon you want to find out possible Istio resource issues. It probably makes sense to use a larger value when your cluster is relatively stable, creation and removal of Istio resource are not very frequent. If your cluster is very dynamic, adding and removing many Istio resources constantly, it probably is better to use a smaller value for the parameter.

Reviewer: ```suggestion
Due to the performance impact when this flag is enabled, you probably can utilize the [`PILOT_ANALYSIS_INTERVAL`](/docs/reference/commands/istioctl/) parameter (available since 1.13.2) to balance how much resource you are willing to spend for validation versus how soon you want to find out possible Istio resource issues. It probably makes sense to use a larger value when your cluster is relatively stable and creation and removal of Istio resource are not very frequent. If your cluster is very dynamic, adding and removing many Istio resources constantly, it probably is better to use a smaller value for the parameter.
```
